### PR TITLE
Avoid unnecessary VeQuickItem::value reads

### PR DESCRIPTION
**Pull request by blammit**
If isValid is true, then the value should be usable.

### CODE REVIEW

File: data/common/Generator.qml [hunk 1]
```diff
@@ -9,11 +9,11 @@ import Victron.VenusOS
 Device {
 	id: generator
 
-	readonly property int state: _state.value === undefined ? -1 : _state.value
+	readonly property int state: _state.isValid ? _state.value : -1
 	readonly property bool autoStart: _autoStart.value === 1
-	readonly property int manualStartTimer: _manualStartTimer.value === undefined ? -1 : _manualStartTimer.value
+	readonly property int manualStartTimer: _manualStartTimer.isValid ? _manualStartTimer.value : 0
```
Comment (DanielMcInnes) [inline data/common/Generator.qml:14]: Intentional behaviour change?

File: components/AcPhase.qml [hunk 2]
```diff
@@ -10,14 +10,11 @@ QtObject {
...
-	readonly property real power: _power.value === undefined ? NaN : _power.value
-	readonly property bool valid: !isNaN(frequency)
-								  && !isNaN(current)
-								  && !isNaN(voltage)
-								  && !isNaN(power)
+	readonly property real frequency: _frequency.isValid ? _frequency.value : NaN
+	readonly property real current: _current.isValid ? _current.value : NaN
+	readonly property real voltage: _voltage.isValid ? _voltage.value : NaN
+	readonly property real power: _power.isValid ? _power.value : NaN
+	readonly property bool valid: _frequency.isValid || _current.isValid || _voltage.isValid || _power.isValid
```
Comment (DanielMcInnes) [inline components/AcPhase.qml:17]: behaviour change?

File: data/common/Generator.qml [hunk 3]
```diff
@@ -9,11 +9,11 @@ import Victron.VenusOS
...
 	readonly property int runtime: _runtime.value || 0
-	readonly property int runningBy: _runningBy.value === undefined ? -1 : _runningBy.value
+	readonly property int runningBy: _runningBy.isValid ? _runningBy.value : 0
```
Comment (DanielMcInnes) [inline data/common/Generator.qml:16]: Intentional behaviour change?

File: data/common/PulseMeter.qml [hunk 4]
```diff
@@ -9,7 +9,7 @@ import Victron.VenusOS
 Device {
 	id: pulseMeter
 
-	readonly property real aggregate: _aggregate.value === undefined ? -1 : _aggregate.value
+	readonly property real aggregate: _aggregate.isValid ? _aggregate.value : NaN
```
Comment (DanielMcInnes) [inline data/common/PulseMeter.qml:12]: Intentional behaviour change?

File: components/AcPhase.qml (hunk 2 above)
Comment (blammit) [reply components/AcPhase.qml:17]: Yes, I think the original intention was to have valid = "any value is valid", so it should be the desired outcome here

File: data/common/Generator.qml (hunk 1 above)
Comment (blammit) [reply data/common/Generator.qml:14]: Yes, the -1 was a bit strange for a "timer" value so I changed it to 0.

File: data/common/Generator.qml (hunk 3 above)
Comment (blammit) [reply data/common/Generator.qml:16]: Yes, similar to the other case, the -1 was a bit strange for a "timer" value so I changed it to 0.

File: data/common/PulseMeter.qml (hunk 4 above)
Comment (blammit) [reply data/common/PulseMeter.qml:12]: Yes, this should have been NaN as -1 doesn't make sense for the real.

File: components/AcPhase.qml (hunk 2 above)
Comment (chriadam) [reply components/AcPhase.qml:17]: the original code seems to require "all values are valid" though?

File: data/common/Generator.qml (hunk 1 above)
Comment (chriadam) [reply data/common/Generator.qml:14]: all uses seem to do: `if (root.generator.manualStartTimer > 0)` so, either 0 or -1 is fine.  I kind of like using -1 for all non-valid integer properties, and NaN for all non-valid real properties, but I have no strong opinion.

**Comment by chriadam** [comment]
Aside from the open question about AcPhase.valid, LGTM.

File: components/AcPhase.qml (hunk 2 above)
Comment (blammit) [reply components/AcPhase.qml:17]: Oh, sorry. Yes, this is a behaviour change, but if a phase only has some of these values set, it seems it should still be considered valid.

Comment (blammit) [reply components/AcPhase.qml:17]: It seems like this `valid` property is not actually used, so I'll remove it.

File: data/common/PulseMeter.qml (hunk 4 above)
Comment (blammit) [reply data/common/PulseMeter.qml:12]: Updated the commit message so that this behavioural change is noted.

File: data/common/Generator.qml (hunk 3 above)
Comment (blammit) [reply data/common/Generator.qml:16]: Updated the commit message so that this behavioural change is noted.

**Review by DanielMcInnes — APPROVED** [review]

**Pull request merged**